### PR TITLE
AS-8430: Update history API to accept timestamp query params

### PR DESCRIPTION
**Pull request by priyakvs**
Alert history API now have the ability to query by timestamp filter. This PR is to add that functionality to cloudflare-go integration as well.

### CODE REVIEW

File: notifications.go [hunk 1]
```diff
@@ -405,19 +405,37 @@ func (api *API) GetAvailableNotificationTypes(ctx context.Context, accountID str
 	return r, nil
 }
 
+// TimeRange is an object for filtering the alert history based on timestamp.
+type TimeRange struct {
+	Since  string `json:"since,omitempty"`
```
Comment (jacobbednarz) [inline notifications.go:410]: something we can do here is apply `url` tags to these fields and instead automatically `Encode` them.

example from `tunnel` - https://github.com/cloudflare/cloudflare-go/blob/fcb3cd41e72cbaeeeb952df36b69d0c4bbff1775/tunnel.go#L105-L113

I might even drop this onto `PaginationOptions` too. WDTY?

**Comment by jacobbednarz** [comment]
i pushed up a change in 19c9713 that swaps to using go-querystring (already in the package) for building and sending the URI parameters to save us manually doing it. do you want to do some quick tests locally with it and confirm it works as you'd expect before we merge this?

**Comment by jacobbednarz** [comment]
all good! i got this working locally using your test and confirmed it on a test account. 

thanks for the PR 🌟

**Comment by priyakvs** [comment]
Thanks a lot for updating this PR @jacobbednarz . Will keep this in mind for all future work :)

**Pull request merged**